components/match2/wikis/clashroyale/match_group_input_custom.lua

```diff
@@ -5,3 +5,597 @@
+
+	return match
+end
+
+function CustomMatchGroupInput._readDate(matchArgs)
+	if matchArgs.date then
+		return MatchGroupInput.readDate(matchArgs.date)
+	else
+		return {
+			date = EPOCH_TIME_EXTENDED,
+			dateexact = false,
+			timestamp = DateExt.epochZero,
+		}
+	end
+end
+
+function CustomMatchGroupInput._checkFinished(match)
+	if Logic.readBoolOrNil(match.finished) == false then
+		match.finished = false
+	elseif Logic.readBool(match.finished) then
+		match.finished = true
+	elseif Logic.isNotEmpty(match.winner) then
+		match.finished = true
+	end
+
+	-- Match is automatically marked finished upon page edit after a
+	-- certain amount of time (depending on whether the date is exact)
+	if match.finished ~= true and match.timestamp > DateExt.epochZero then
+		local currentUnixTime = os.time(os.date('!*t'))
+		local threshold = match.dateexact and 30800 or 86400
+		if match.timestamp + threshold < currentUnixTime then
+			match.finished = true
+		end
+	end
+
+	return match
+end
+
+function CustomMatchGroupInput._getTournamentVars(match)
+	match.mode = Logic.emptyOr(match.mode, Variables.varDefault('tournament_mode', 'solo'))
+	match.publishertier = Logic.emptyOr(match.publishertier, Variables.varDefault('tournament_publishertier'))
+	return MatchGroupInput.getCommonTournamentVars(match)
+end
+
+function CustomMatchGroupInput._getVodStuff(match)
+	match.stream = Streams.processStreams(match)
+	match.vod = Logic.emptyOr(match.vod)
+
+	return match
+end
+
+function CustomMatchGroupInput._getLinks(match)
+	match.links = {
+		preview = match.preview,
+		preview2 = match.preview2,
+		interview = match.interview,
+		interview2 = match.interview2,
+		review = match.review,
+		recap = match.recap,
+	}
+	return match
+end
+
+function CustomMatchGroupInput._getExtraData(match)
+	local extradata = {
+		casters = match.casters,
+		t1bans = CustomMatchGroupInput._readBans(match.t1bans),
+		t2bans = CustomMatchGroupInput._readBans(match.t2bans),
+	}
+
+	match.extradata = extradata
+
+	return match
+end
+
+function CustomMatchGroupInput._readBans(bansInput)
+	local bans = CustomMatchGroupInput._readCards(bansInput)
+
+	return Table.isNotEmpty(bans) and bans or nil
+end
+
+function CustomMatchGroupInput._adjustData(match)
+	--parse opponents + set base sumscores
+	match = CustomMatchGroupInput._opponentInput(match)
+
+	--main processing done here
+	local subGroupIndex = 0
+	for _, _, mapIndex in Table.iter.pairsByPrefix(match, 'map') do
+		match, subGroupIndex = CustomMatchGroupInput._mapInput(match, mapIndex, subGroupIndex)
+	end
+
+	match = CustomMatchGroupInput._matchWinnerProcessing(match)
+
+	if match.hasTeamOpponent then
+		error('Team matches not yet supported')
+	end
+
+	if Logic.isNumeric(match.winner) then
+		match.finished = true
+	end
+
+	return match
+end
+
+--[[
+
+Misc. MatchInput functions
+--> Winner, Walkover, Placement, Resulttype, Status
+
+]]--
+function CustomMatchGroupInput._matchWinnerProcessing(match)
```

Comment:
What's the point of this function + `_checkFinished` + L144 (if winner -> finished = true)?
Seems like they are all trying to do the same thing (setting finished), but have different and contraindicatory logic.

Reply:
L144 basically catches manual winner input (could probably merge it into this function
`_checkFinished` only does the time check to determine if a match should be marked as finished even though it does not (yet) have a winner

could probably clean it up a bit and merge stuff

might be able to look into it tomorrow but not today